feat(atomic): string serialization of atoms

jpnelson opened the pull request:
<!--
Thanks for submitting a pull request!
Please provide enough information so that others can review your pull request.
Motivation and Test plan are mandatory.
-->

## Motivation

Addresses https://github.com/callstack/linaria/issues/912

## Summary

Should be fairly straightforward after reading the reason why we might want this. I've also updated the documentation to reflect what's going.

## Test plan

Tested with snapshot tests and unit tests. Docs should render the same way

<!--
Demonstrate the code is solid.
Example: The exact commands you ran and their output.
Write down steps on how maintainers can test your PR.
-->

jpnelson commented:
Hi @Anber , thanks so much for looking at the other PR's I've put up!

This one addresses some issues we've come across when using atomic styles in Linaria – namely, it's confusing for people that `cx(a, cx(b, c)) !== cx(a, b, c)`. This should fix that by just providing strings. Another advantage is you don't (strictly) need to use cx with atomic styles if you're not merging – you could apply the class string directly.

Would love to get your thoughts! The issue linked provides more detailed rationale. Thanks again!

Anber commented on packages/babel/tsconfig.json:
```diff
@@ -2,6 +2,7 @@
   "extends": "../../tsconfig.json",
   "compilerOptions": { "paths": {}, "rootDir": "src/" },
   "references": [
+    { "path": "../atomic" },
```
Looks like it's unneeded here.

jpnelson replied:
Thanks! Removed

jpnelson replied:
I realized this is being added because there's a dev dependency `@linaria/atomic`. We can keep it removed it but I think that's why it's being re-added.

Anber commented:
Hi @jpnelson!
It looks good to me but I don't have enough experience with atomic styles, so all decisions about it are up to you :)

jpnelson commented:
@Anber I added another commit to this PR, to add property priorities to address https://github.com/callstack/linaria/issues/902 . It depends on this PR so I added it to this one.

Do you think you'd be able to do a release when this goes in? (I think we'd need new versions of `@linaria/babel-preset` and `@linaria/atomic`. Thanks again for your reviews!

jpnelson commented:
@Anber sorry to ping again on this – do you think a merge / release is possible?

We're also testing and iterating on these features in our project. I'm hopeful this is the last major change, but if the frequency of releases is inconvenient, we could create a fork for our project and try and contribute back later in a larger chunk.

Thanks again for your time! 🙏

Anber commented:
@jpnelson sorry for the delay, I was in mountains for the last two weeks almost without all communications. I'll merge it and release in a couple of days.

jpnelson commented:
Haha that's very understandable, thank you! 🙏

Anber commented on packages/atomic/src/atomize.ts:
```diff
@@ -1,6 +1,26 @@
 import { slugify } from '@linaria/utils';
 import stylis from 'stylis';
+import { all as knownProperties } from 'known-css-properties';
+import { getPropertyPriority } from './propertyPriority';
```
Hi @jpnelson
Looks like you didn't add this file to the PR

jpnelson replied:
my mistake, sorry @Anber ! I've added it back in https://github.com/callstack/linaria/pull/950 . I think this was missing because it was originally a separate PR.

Sorry I missed that too – the precommit hooks seem extensive, but let me know if there's anything else I can run before merging – CI doesn't seem to start for me when I create PR's.
